x/bridge/observer/observer_test.go

```diff
@@ -2,125 +2,137 @@ package observer_test
 
 import (
 	"context"
-	"encoding/json"
-	"net/http"
-	"net/http/httptest"
-	"os"
+	"sync/atomic"
 	"testing"
 	"time"
 
-	abcitypes "github.com/cometbft/cometbft/abci/types"
-	cmtjson "github.com/cometbft/cometbft/libs/json"
+	"cosmossdk.io/math"
 	"github.com/cometbft/cometbft/libs/log"
-	coretypes "github.com/cometbft/cometbft/rpc/core/types"
-	"github.com/cometbft/cometbft/rpc/jsonrpc/types"
-	cmttypes "github.com/cometbft/cometbft/types"
-	proto "github.com/cosmos/gogoproto/proto"
-	"github.com/gorilla/websocket"
 	"github.com/stretchr/testify/require"
 
 	"github.com/osmosis-labs/osmosis/v24/x/bridge/observer"
-	bridge "github.com/osmosis-labs/osmosis/v24/x/bridge/types"
 )
 
-var upgrader = websocket.Upgrader{}
+type MockChain struct {
+	In  chan observer.Transfer
+	Out chan observer.Transfer
+	H   atomic.Uint64
+	CR  atomic.Uint64
```

Comment:
Lets please go for the more verbose "confirmationsRequired" and "height"

Reply:
Done in https://github.com/osmosis-labs/osmosis/pull/7882